app/extensions/api/namespace.py

```diff
@@ -170,6 +175,9 @@ def wrapper(*args, **kwargs):
 
                 protected_func = _permission_decorator(func)
 
+            if not hasattr(protected_func, '_decorators'):
```

Comment:
Please, create a separate function to avoid such XXX code duplication. I'm still in doubt if I advised a right way of doing this (I mean using `_decorators` attribute on the function itself).